src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutMissingTagCheck.java

```diff
@@ -0,0 +1,169 @@
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.AreaTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.ServiceTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Angle;
+import org.openstreetmap.atlas.utilities.tuples.Tuple;
+
+/**
+ * Flags edges that are closed and round shaped without junction=roundabout tag and have minimum of
+ * two intersections with navigable roads {@link RoundaboutMissingTagCheck#MINIMUM_INTERSECTION}
+ * connections. See https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout for more
+ * information about roundabouts
+ * 
+ * @author vladlemberg
+ */
+
+public class RoundaboutMissingTagCheck extends BaseCheck<Long>
+{
+    // Instructions
+    public static final String MISSING_JUNCTION_TAG_INSTRUCTION = "This edge might be a roundabout";
+    // Minimum intersection with Navigable Roads
+    private static final int MINIMUM_INTERSECTION = 2;
+    private static final int MODULUS = 10;
+    private static final int FIRST_EDGE_SECTION = 1;
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(MISSING_JUNCTION_TAG_INSTRUCTION);
+    private static final double MAX_THRESHOLD_DEGREES_DEFAULT = 40.0;
+    private static final double MIN_THRESHOLD_DEGREES_DEFAULT = 10.0;
+    private final Angle maxAngleThreshold;
+    private final Angle minAngleThreshold;
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public RoundaboutMissingTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.maxAngleThreshold = configurationValue(configuration, "angle.threshold.maximum_degree",
+                MAX_THRESHOLD_DEGREES_DEFAULT, Angle::degrees);
+        this.minAngleThreshold = configurationValue(configuration, "angle.threshold.minimum_degree",
+                MIN_THRESHOLD_DEGREES_DEFAULT, Angle::degrees);
+    }
+
+    /**
+     * Validates if given {@link AtlasObject} is actually an {@link Edge} and is a potential
+     * roundabout and also corresponding OSM identifier shouldn't be flagged before (this is for
+     * avoiding duplicate flags)
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && !this.isFlagged(object.getOsmIdentifier())
+                && object.getIdentifier() % MODULUS == FIRST_EDGE_SECTION
+                && ((Edge) object).isMasterEdge() && HighwayTag.isCarNavigableHighway(object)
+                && object.getTag(JunctionTag.KEY).isEmpty() && object.getTag(AreaTag.KEY).isEmpty()
+                && isPartOfClosedWay((Edge) object) && intersectingWithMoreThan((Edge) object);
```

Comment:
nit: **this**.isPartOfClosedWay((Edge) object) && **this**.intersectingWithMoreThan((Edge) object)